MapboxCoreNavigation/Geometry.swift

```diff
@@ -166,11 +166,10 @@ func closestCoordinate(on polyline: [CLLocationCoordinate2D], to coordinate: CLL
             closestCoordinate = CoordinateAlongPolyline(coordinate: segment.0, index: index, distance: distances.0)
         }
         if distances.1 < closestCoordinate?.distance ?? CLLocationDistanceMax {
-            index = index + 1
-            closestCoordinate = CoordinateAlongPolyline(coordinate: segment.1, index: index, distance: distances.1)
+            closestCoordinate = CoordinateAlongPolyline(coordinate: segment.1, index: index+1, distance: distances.1)
         }
         if intersectionDistance != nil && intersectionDistance! < closestCoordinate?.distance ?? CLLocationDistanceMax {
-            closestCoordinate = CoordinateAlongPolyline(coordinate: intersectionPoint!, index: index, distance: intersectionDistance!)
+            closestCoordinate = CoordinateAlongPolyline(coordinate: intersectionPoint!, index: (distances.0 < distances.1 ? index : index+1), distance: intersectionDistance!)
```

Comment:
> My first instinct is that adding `index + 1` will in fact return _the next_ closest coordinate and not the actual closest coordinate.

@bsudekum, this change can be seen as completing the work begun in #39, making the index more consistent with the returned coordinate. `index` initially points to the vertex at the beginning of the segment being examined, so if the coordinate is closer to the end of the segment, the closest vertex would be at `index + 1`.

> FWIW, I believe at one point this function did accept an optional `look next point?` argument.

You’re probably thinking of `includeDistanceToNextCoordinate`, which was removed in #9. In addition to conditionally incrementing the `index` in this third case, the `includeDistanceToNextCoordinate` argument also [updated the `intersectionDistance`](https://github.com/mapbox/mapbox-navigation-ios/blob/5862a33bf76a0247a144c0a4a5f599bd10bb5464/MapboxNavigation/Geometry.swift#L151-L156) (incorrectly).